### ISSUE TITLE
[CP] fullscreenDialog transition is broken in 3.27.1

### ISSUE THREAD

**Issue by RepliedSage11** [flutter/flutter#160420]
### Issue Link

https://github.com/flutter/flutter/issues/159289

### Target

stable

### Cherry pick PR Link

https://github.com/flutter/flutter/pull/159312

### Changelog Description

When pushing a route with fullscreenDialog, the underlying route is transitioned out to the left side while the fullscreen route is pushed from bottom up. This leads to a broken transition animation.

### Impacted Users

all Flutter devs

### Impact Description

Visual glitch noticeable to the end-users

### Workaround

Not to my knowledge

### Risk

low

### Test Coverage

yes

### Validation Steps

Run the steps to reproduce from the original issue

**Comment by christopherfujino** [comment]
@justinmc WDYT about this? does this warrant a stable CP?

**Comment by justinmc** [comment]
Yes I think it should probably be cherry picked. All apps using `fullscreenDialog: true` will probably encounter this, and it looks pretty glitchy even if it doesn't prevent users from using the app.

**Comment by MitchellGoodwin** [comment]
Yes, that PR gets rid of visual jank introduced by 3.27 and the fix is very low risk.

**Comment by RepliedSage11** [comment]
@christopherfujino Hi, is this expected to land in 3.27.2?

**Comment by RepliedSage11** [comment]
@justinmc @MitchellGoodwin sorry for bothering, but is there an approximate timeframe for when this can be cherry-picked to the stable release? This is a blocker for us.

**Comment by christopherfujino** [comment]
> [@christopherfujino[<img alt="" width="15" height="15" src="chrome-extension://dlebflppeeemcdpidccbiblndppbmjmh/ospo-chrome-ext-logo.png">](https://teams.googleplex.com/fujino@google.com)](https://github.com/christopherfujino) Hi, is this expected to land in 3.27.2?

Sorry, I messed up when I made the 3.27.2 release, and skipped this. It will be in the next release, which we're working on presently.

**Comment by yuukiw00w** [comment]
The original issue (#159289) mentions the following, but it was released in a broken state:

> @MitchellGoodwin Could you take a look at this to make sure we don't release this in a broken state? Could be related to #150031.

#159289 is a P1-labeled, high-priority bug, and a prompt fix is highly desired.
At the very least, I hope the fix will be included in the next release.
Thank you for your attention.

**Comment by eyebrowsoffire** [comment]
This fix is in 3.27.3 stable release
